### PR TITLE
Tools 108 filter atac ids and update to concat

### PR DESCRIPTION
**Pull request by jychien**
@jahilton: Things updated:
- replace append() with concat()
- make sure var index is same size as one of raw matrices
- convert author_cell_type_column datatype if needed
- atac gene activity matrices will filter ensembl IDs for appropriate annotation version

I actually don't make sure that the atac processed matrix file is indexed by ensembl. Should I make this assumption? Or do a check?

### CODE REVIEW

**Comment by jahilton** [comment]
>I actually don't make sure that the atac processed matrix file is indexed by ensembl. Should I make this assumption? Or do a check?

I think you can assume that. It's something that we can build into checkfiles now that we have enough consistency to run these files through.

File: scripts/flattener.py [hunk 1]
```diff
@@ -629,22 +633,15 @@ def set_ensembl(cxg_adata, cxg_adata_raw, redundant, feature_keys):
 		unique_to_norm =  set(cxg_adata.var.index.to_list()).difference(set(cxg_adata_raw.var.index.to_list()))
 		if len(unique_to_norm) > 0:
 			print("WARNING: normalized matrix contains Ensembl IDs not in raw: {}".format(unique_to_norm))
+	del cxg_adata_raw.var['feature_biotype'] # remove feature_biotype from var and raw.var
+	del cxg_adata.var['feature_biotype']
```
Comment (jahilton) [inline scripts/flattener.py:637]: can we prevent these (`feature_biotype`) from being added in the first place?

Comment (jychien) [reply scripts/flattener.py:637]: Yes, I want to clean up feature_biotype, but I was thinking to do it in a separate ticket to break up the work. I can make another ticket for that and also move more code from main() to separate functions.

Comment (jahilton) [reply scripts/flattener.py:637]: works for me

**Review by jahilton — APPROVED** [review]

**Pull request merged**